### PR TITLE
fix: do not scan local modules

### PR DESCRIPTION
**Pull request by thisislawatts**
Do not treat local modules as if they are 3rd party packages hosted on npm.

Reference: #11 

This PR addresses the above need in two ways:

* Validates that a folder/file does not exist locally before assuming it is an external package.
* Introduces a new configuration option to ignore paths based on a Regular Expression

### CODE REVIEW

File: src/getImports/babelParser.js [hunk 1]
```diff
@@ -26,13 +30,39 @@ export function getPackages(fileName, source, language) {
...
+
+      if (pathIgnored) {
+        logger.log(`Import ${node.source.value} matched ignored path: ${path}`);
+        return;
+      }
+
+      const target = path.dirname(fileName) + path.sep + node.source.value;
+      const fileExists =
+        fs.existsSync(target) ||
+        fs.existsSync(target + '.js') ||
```
Comment (erwinw) [inline src/getImports/babelParser.js]: Please make the extensions configurable - think of `.jsx`, `.ts`, etc. Thanks!

File: src/getImports/babelParser.js [hunk 2]
```diff
@@ -26,13 +30,39 @@ export function getPackages(fileName, source, language) {
...
-      });
+      const configuration = workspace.getConfiguration('vulnCost');
+      let pathIgnored = false;
+
+      for (let i = 0; i < configuration.ignorePaths.length; i++) {
+        const path = configuration.ignorePaths[i];
+        pathIgnored = new RegExp(path).test(node.source.value);
+
+        if (pathIgnored) {
+          continue;
```
Comment (erwinw) [inline src/getImports/babelParser.js]: Isn’t the intention to break out of the `for` loop here?
```suggestion
          break;
```

File: src/getImports/babelParser.js (hunk 1 above)
Comment (thisislawatts) [reply src/getImports/babelParser.js]: Excellent point @erwinpleo, we already expose `typescriptExtensions` and `javascriptExtensions` as configuration options so we should reuse these

Comment (thisislawatts) [reply src/getImports/babelParser.js]: Proposed implementation here:
https://github.com/snyk/vulncost/pull/16/commits/927e53ccbc5a4213ac3057d718a9bcda80b245cb

**Review by bmvermeer — APPROVED** [review]
Works great!

**Comment by bmvermeer** [comment]
need to add entry to changelog and merge

**Pull request merged**